### PR TITLE
[VEN-2797]: Grant and Revoke Timelock Permissions

### PR DESCRIPTION
**Pull request by web3rover**
## Description

Resolves VEN

### CODE REVIEW

File: multisig/proposals/arbitrumsepolia/vip-014/index.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+      target: arbitrumsepolia.VTREASURY,
+      signature: "transferOwnership(address)",
+      params: [arbitrumsepolia.NORMAL_TIMELOCK],
+    },
+    {
+      target: ACM,
+      signature: "grantRole(bytes32,address)",
+      params: [DEFAULT_ADMIN_ROLE, arbitrumsepolia.NORMAL_TIMELOCK],
+    },
+  ]);
```
Comment (GitGuru7) [inline multisig/proposals/arbitrumsepolia/vip-014/index.ts:139]: We don't need this command on all the networks, its already done here
https://github.com/VenusProtocol/vips/blob/main/multisig/proposals/arbitrumsepolia/vip-009/index.ts#L114

File: simulations/vip-373/arbitrumsepolia.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import ACCESS_CONTROL_MANAGER_ABI from "./abi/AccessControlManager.json";
+
+forking(86016842, async () => {
+  before(async () => {
+    await pretendExecutingVip(await vip014());
+  });
+
+  testForkedNetworkVipCommands("vip373", await vip373(), {
+    callbackAfterExecution: async txResponse => {
+      await expectEvents(txResponse, [ACCESS_CONTROL_MANAGER_ABI], ["PermissionGranted"], [84]);
```
Comment (GitGuru7) [inline simulations/vip-373/arbitrumsepolia.ts:15]: Can we also check for GrantPermissionsExecuted and RevokePermissionsExecuted events?

File: simulations/vip-373/sepolia.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  before(async () => {
+    await pretendExecutingVip(await vip060());
+  });
+
+  testForkedNetworkVipCommands("vip373", await vip373(), {
+    callbackAfterExecution: async txResponse => {
+      await expectEvents(txResponse, [ACCESS_CONTROL_MANAGER_ABI], ["PermissionGranted"], [95]);
+      await expectEvents(txResponse, [ACCESS_CONTROL_MANAGER_ABI], ["PermissionRevoked"], [42]);
+    },
+  });
```
Comment (GitGuru7) [inline simulations/vip-373/sepolia.ts:18]: Can we also verify the roles in post VIP? Something like this 
https://github.com/VenusProtocol/vips/blob/main/multisig/simulations/arbitrumsepolia/vip-009/index.ts#L38
Just to ensure everything is working fine.

**Review by GitGuru7 — COMMENTED** [review]
Can we merge vip-372 and vip-373 so reduce one VIP, WDYT?

File: simulations/vip-373/sepolia.ts (hunk 3 above)
Comment (GitGuru7) [reply simulations/vip-373/sepolia.ts:23]: Not all, but few of them like 2-3 permissions.

File: multisig/proposals/arbitrumone/vip-014/index.ts [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+export const XVS_BRIDGE_ADMIN_PROXY = "0xf5d81C6F7DAA3F97A6265C8441f92eFda22Ad784";
+export const XVS = "0xc1Eb7689147C81aC840d4FF0D298489fc7986d52";
+export const BOUND_VALIDATOR = "0x2245FA2420925Cd3C2D889Ddc5bA1aefEF0E14CF";
+
+export const VTOKENS = [
+  "0xaDa57840B372D4c28623E87FC175dE8490792811",
+  "0x68a34332983f4Bf866768DD6D6E638b02eF5e1f0",
+  "0xB9F9117d4200dC296F9AcD1e8bE1937df834a2fD",
+  "0x7D8609f8da70fF9027E9bc5229Af4F6727662707",
+  "0xAeB0FEd69354f34831fe1D16475D9A83ddaCaDA6",
```
Comment (chechu) [inline multisig/proposals/arbitrumone/vip-014/index.ts:26]: We have new markets on Arbitrum one (LST pool)
```suggestion
  "0xAeB0FEd69354f34831fe1D16475D9A83ddaCaDA6",
  "0x39D6d13Ea59548637104E40e729E4aABE27FE106",
  "0x246a35E79a3a0618535A469aDaF5091cAA9f7E88",
  "0x9df6B5132135f14719696bBAe3C54BAb272fDb16",
```

File: multisig/proposals/arbitrumsepolia/vip-014/index.ts [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+export const XVS_STORE = "0x4e909DA6693215dC630104715c035B159dDb67Dd";
+export const XVS_BRIDGE_ADMIN_PROXY = "0xc94578caCC89a29B044a0a1D54d20d48A645E5C8";
+export const XVS = "0x877Dc896e7b13096D3827872e396927BbE704407";
+
+export const VTOKENS = [
+  "0x49FB90A5815904649C44B87001a160C1301D6a2C",
+  "0x807dCB6946dDF4C5C6446B1B07ACd248B08F45e2",
+  "0xdEFbf0F9Ab6CdDd0a1FdDC894b358D0c0a39B052",
+  "0xd9d1e754464eFc7493B177d2c7be04816E089b4C",
+  "0x292Ec2b45C549Bc2c6B31937dBd511beaAEabea8",
```
Comment (chechu) [inline multisig/proposals/arbitrumsepolia/vip-014/index.ts:25]:
```suggestion
  "0x292Ec2b45C549Bc2c6B31937dBd511beaAEabea8",
  "0xd7057250b439c0849377bB6C3263eb8f9cf49d98",
  "0x75f841b14305935D8D7E806f249D9FA52EF1550B",
  "0x253515E19e8b888a4CA5a0a3363B712402ce4046",
```

File: multisig/proposals/ethereum/vip-061/index.ts [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+  "0x7c8ff7d2A1372433726f879BD945fFb250B94c65",
+  "0x2d499800239C4CD3012473Cb1EAE33562F0A6933",
+  "0x30aD10Bd5Be62CAb37863C2BfcC6E8fb4fD85BDa",
+  "0xA854D35664c658280fFf27B6eDC6C4195c3229B3",
+  "0x76697f8eaeA4bE01C678376aAb97498Ee8f80D5C",
+  "0xDB6C345f864883a8F4cae87852Ac342589E76D1B",
+  "0xF9E9Fe17C00a8B96a8ac20c4E344C8688D7b947E",
+  "0xc82780Db1257C788F262FBbDA960B3706Dfdcaf2",
+  "0x4a240F0ee138697726C8a3E43eFE6Ac3593432CB",
+  "0xb4933AF59868986316Ed37fa865C829Eba2df0C7",
```
Comment (chechu) [inline multisig/proposals/ethereum/vip-061/index.ts:61]: VToken_vweETHs_LiquidStakedETH:

```suggestion
  "0xb4933AF59868986316Ed37fa865C829Eba2df0C7",
  "0xEF26C64bC06A8dE4CA5D31f119835f9A1d9433b9",
```

File: multisig/proposals/sepolia/vip-060/index.ts [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+  "0xF4C1B7528f8B266D8ADf1a85c91d93114FeDbA2A",
+  "0x3AF2bE7AbEF0f840b196D99d79F4B803a5dB14a1",
+  "0x20a83DE526F2CF2fCec2131E07b11F956d8f3Cdf",
+  "0x83F63118dcAAdAACBFF36D78ffB88dd474309e70",
+  "0x9f6213dFa9069a5426Fe8fAE73857712E1259Ed4",
+  "0x0a95088403229331FeF1EB26a11F9d6C8E73f23D",
+  "0x30c31bA6f4652B548fe7a142A949987c3f3Bf80b",
+  "0x9C5e7a3B4db931F07A6534f9e44100DDDc78c408",
+  "0xD5f83FCbb4a62779D0B37b9E603CD19Ad84884F0",
+  "0x93dff2053D4B08823d8B39F1dCdf8497f15200f4",
```
Comment (chechu) [inline multisig/proposals/sepolia/vip-060/index.ts:57]: VToken_vweETHs_LiquidStakedETH:
```suggestion
  "0x93dff2053D4B08823d8B39F1dCdf8497f15200f4",
  "0xB3A201887396F57bad3fF50DFd02022fE1Fd1774",
```

File: vips/vip-372/bscmainnet.ts [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+        signature: "acceptOwnership()",
+        params: [],
+        dstChainId: LzChainId.opbnbmainnet,
+      },
+      {
+        target: ethereum.VTREASURY,
+        signature: "acceptOwnership()",
+        params: [],
+        dstChainId: LzChainId.ethereum,
+      },
```
Comment (chechu) [inline vips/vip-372/bscmainnet.ts:180]: TODO: add commands to grant and revoke permissions using the ACMCommandsAggregator contract

File: simulations/vip-371/sepolia.ts [hunk 9]
```diff
@@ -0,0 +1,49 @@
+import { expect } from "chai";
+import { ethers } from "hardhat";
+import { NETWORK_ADDRESSES } from "src/networkAddresses";
+import { expectEvents } from "src/utils";
+import { forking, pretendExecutingVip, testForkedNetworkVipCommands } from "src/vip-framework";
+
+import vip060, { ACM } from "../../multisig/proposals/sepolia/vip-060";
+import vip371, { DEFAULT_ADMIN_ROLE, OPBNBTESTNET_ACM_AGGREGATOR } from "../../vips/vip-371/bsctestnet";
```
Comment (chechu) [inline simulations/vip-371/sepolia.ts:8]:
```suggestion
import vip371, { DEFAULT_ADMIN_ROLE, SEPOLIA_ACM_AGGREGATOR } from "../../vips/vip-371/bsctestnet";
```

File: simulations/vip-371/sepolia.ts [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+      await expectEvents(txResponse, [ACCESS_CONTROL_MANAGER_ABI], ["PermissionRevoked"], [67]);
+      await expectEvents(txResponse, [ACM_COMMANDS_AGGREGATOR_ABI], ["GrantPermissionsExecuted"], [1]);
+      await expectEvents(txResponse, [ACM_COMMANDS_AGGREGATOR_ABI], ["RevokePermissionsExecuted"], [1]);
+    },
+  });
+
+  describe("Post-VIP behaviour", async () => {
+    const acm = new ethers.Contract(ACM, ACCESS_CONTROL_MANAGER_ABI, ethers.provider);
+    it("check if DEFAULT_ROLE has been revoked for ACMAggregator", async () => {
+      expect(await acm.hasRole(DEFAULT_ADMIN_ROLE, OPBNBTESTNET_ACM_AGGREGATOR)).to.be.false;
```
Comment (chechu) [inline simulations/vip-371/sepolia.ts:31]:
```suggestion
      expect(await acm.hasRole(DEFAULT_ADMIN_ROLE, SEPOLIA_ACM_AGGREGATOR)).to.be.false;
```

File: simulations/vip-371/arbitrumsepolia.ts [hunk 11]
```diff
@@ -0,0 +1,50 @@
+import { expect } from "chai";
+import { ethers } from "hardhat";
+import { NETWORK_ADDRESSES } from "src/networkAddresses";
+import { expectEvents } from "src/utils";
+import { forking, pretendExecutingVip, testForkedNetworkVipCommands } from "src/vip-framework";
+
+import vip014, { ACM } from "../../multisig/proposals/arbitrumsepolia/vip-014";
+import vip371, { DEFAULT_ADMIN_ROLE, OPBNBTESTNET_ACM_AGGREGATOR } from "../../vips/vip-371/bsctestnet";
```
Comment (chechu) [inline simulations/vip-371/arbitrumsepolia.ts:8]:
```suggestion
import vip371, { DEFAULT_ADMIN_ROLE, ARBITRUMSEPOLIA_ACM_AGGREGATOR } from "../../vips/vip-371/bsctestnet";
```

File: simulations/vip-371/arbitrumsepolia.ts [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+      await expectEvents(txResponse, [ACM_COMMANDS_AGGREGATOR_ABI], ["GrantPermissionsExecuted"], [1]);
+      await expectEvents(txResponse, [ACM_COMMANDS_AGGREGATOR_ABI], ["RevokePermissionsExecuted"], [1]);
+    },
+  });
+
+  describe("Post-VIP behaviour", async () => {
+    const acm = new ethers.Contract(ACM, ACCESS_CONTROL_MANAGER_ABI, ethers.provider);
+
+    it("check if DEFAULT_ROLE has been revoked for ACMAggregator", async () => {
+      expect(await acm.hasRole(DEFAULT_ADMIN_ROLE, OPBNBTESTNET_ACM_AGGREGATOR)).to.be.false;
```
Comment (chechu) [inline simulations/vip-371/arbitrumsepolia.ts:32]:
```suggestion
      expect(await acm.hasRole(DEFAULT_ADMIN_ROLE, ARBITRUMSEPOLIA_ACM_AGGREGATOR)).to.be.false;
```

File: multisig/proposals/arbitrumsepolia/vip-014/index.ts [hunk 13]
```diff
@@ -0,0 +1,145 @@
...
+  "0xd7057250b439c0849377bB6C3263eb8f9cf49d98",
+  "0x75f841b14305935D8D7E806f249D9FA52EF1550B",
+  "0x253515E19e8b888a4CA5a0a3363B712402ce4046",
+];
+export const BOUND_VALIDATOR = "0xfe6bc1545Cc14C131bacA97476D6035ffcC0b889";
+
+export const DEFAULT_ADMIN_ROLE = "0x0000000000000000000000000000000000000000000000000000000000000000";
```
Comment (GitGuru7) [inline multisig/proposals/arbitrumsepolia/vip-014/index.ts:32]:
```suggestion
```

File: multisig/proposals/ethereum/vip-061/index.ts [hunk 14]
```diff
@@ -0,0 +1,205 @@
...
+  "0xEF26C64bC06A8dE4CA5D31f119835f9A1d9433b9",
+];
+export const XVS_STORE = "0x1Db646E1Ab05571AF99e47e8F909801e5C99d37B";
+export const XVS_BRIDGE_ADMIN_PROXY = "0x9C6C95632A8FB3A74f2fB4B7FfC50B003c992b96";
+export const XVS = "0xd3CC9d8f3689B83c91b7B59cAB4946B063EB894A";
+export const BOUND_VALIDATOR = "0x1Cd5f336A1d28Dff445619CC63d3A0329B4d8a58";
+export const SFrxETHOracle = "0x5E06A5f48692E4Fff376fDfCA9E4C0183AAADCD1";
+
+export const DEFAULT_ADMIN_ROLE = "0x0000000000000000000000000000000000000000000000000000000000000000";
```
Comment (GitGuru7) [inline multisig/proposals/ethereum/vip-061/index.ts:70]:
```suggestion
```

**Review by GitGuru7 — APPROVED** [review]

File: simulations/vip-371/arbitrumsepolia.ts [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+import vip371, { DEFAULT_ADMIN_ROLE, ARBITRUMSEPOLIA_ACM_AGGREGATOR } from "../../vips/vip-371/bsctestnet";
+import ACM_COMMANDS_AGGREGATOR_ABI from "./abi/ACMCommandsAggregator.json";
+import ACCESS_CONTROL_MANAGER_ABI from "./abi/AccessControlManager.json";
+
+const { arbitrumsepolia } = NETWORK_ADDRESSES;
+
+forking(87457288, async () => {
+  before(async () => {
+    await pretendExecutingVip(await vip014());
+  });
```
Comment (chechu) [inline simulations/vip-371/arbitrumsepolia.ts:17]: I think we don't need this step in this simulation

File: simulations/vip-371/opbnbtestnet.ts [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+import vip371, { DEFAULT_ADMIN_ROLE, OPBNBTESTNET_ACM_AGGREGATOR } from "../../vips/vip-371/bsctestnet";
+import ACM_COMMANDS_AGGREGATOR_ABI from "./abi/ACMCommandsAggregator.json";
+import ACCESS_CONTROL_MANAGER_ABI from "./abi/AccessControlManager.json";
+
+const { opbnbtestnet } = NETWORK_ADDRESSES;
+
+forking(41684455, async () => {
+  before(async () => {
+    await pretendExecutingVip(await vip021());
+  });
```
Comment (chechu) [inline simulations/vip-371/opbnbtestnet.ts:17]: I think we don't need this step in this simulation

File: simulations/vip-371/sepolia.ts [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+import vip371, { DEFAULT_ADMIN_ROLE, SEPOLIA_ACM_AGGREGATOR } from "../../vips/vip-371/bsctestnet";
+import ACM_COMMANDS_AGGREGATOR_ABI from "./abi/ACMCommandsAggregator.json";
+import ACCESS_CONTROL_MANAGER_ABI from "./abi/AccessControlManager.json";
+
+const { sepolia } = NETWORK_ADDRESSES;
+
+forking(6850243, async () => {
+  before(async () => {
+    await pretendExecutingVip(await vip060());
+  });
```
Comment (chechu) [inline simulations/vip-371/sepolia.ts:17]: I think we don't need this step in this simulation

File: vips/vip-371/bsctestnet.ts [hunk 18]
```diff
@@ -0,0 +1,104 @@
...
+export const ARBITRUMSEPOLIA_ACM_AGGREGATOR = "0x4fCbfE445396f31005b3Fd2F6DE2A986d6E2dCB5";
+export const OPBNBTESTNET_ACM_AGGREGATOR = "0xbDd501dB1B0D6aab299CE69ef5B86C8578947AD0";
+export const SEPOLIA_ACM_AGGREGATOR = "0x0653830c55035d678e1287b2d4550519fd263d0e";
+
+export const DEFAULT_ADMIN_ROLE = "0x0000000000000000000000000000000000000000000000000000000000000000";
+
+const vip371 = () => {
+  const meta = {
+    version: "v2",
+    title: "VIP-332 accept ownership & give permissions to Normal Timelock",
```
Comment (chechu) [inline vips/vip-371/bsctestnet.ts:17]:
```suggestion
    title: "VIP-371 grant Timelocks and revoke permissions for Guardian in Arbitrum, opBNB and Ethereum",
```

File: vips/vip-371/bscmainnet.ts [hunk 19]
```diff
@@ -0,0 +1,110 @@
...
+        signature: "executeGrantPermissions(uint256)",
+        params: [0],
+        dstChainId: LzChainId.ethereum,
+      },
+      {
+        target: ETHEREUM_ACM_AGGREGATOR,
+        signature: "executeGrantPermissions(uint256)",
+        params: [1],
+        dstChainId: LzChainId.ethereum,
+      },
```
Comment (chechu) [inline vips/vip-371/bscmainnet.ts:92]: Do we need to execute both batches on Ethereum?

**Review by chechu — APPROVED** [review]

**Comment by chechu** [comment]
https://app.venus.io/#/governance/proposal/391?chainId=56

**Pull request merged**